### PR TITLE
Don't let child process become zombie

### PR DESCRIPTION
**Pull request by prashanthpai**
glusterfsd, when spawned will exit because:
* Directories in logfile path and pidfile path don't exist
* It tries to use portmap and communicate with glusterd to get volfile.
  This isn't implemented yet.

When glusterfsd dies immediately after spawn, it'll become zombie if
exit status of the process isn't read by parent.

This change also moves the code that reads pid file to place where it
really should be. Also checks if process is really running before it
returns success to the caller.

Signed-off-by: Prashanth Pai <ppai@redhat.com>

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -64,16 +74,27 @@ func Start(d Daemon, wait bool) error {
 		// Wait for the process to exit
 		err = cmd.Wait()
 		return err
+	} else {
+		// If the process exits at some point later, do read it's
+		// exit status. This should not let it be a zombie.
+		done := make(chan error, 1)
+		go func() {
+			done <- cmd.Wait()
```
Comment (kshlm) [inline daemon/daemon.go]: You don't need a channel here. You can just directly assign to `err`. As a thumb rule, you should be using channels when communicating across channels most of the time.

File: daemon/daemon.go [hunk 2]
```diff
@@ -64,16 +74,27 @@ func Start(d Daemon, wait bool) error {
...
-	if err == nil {
-		// Check if process is running
-		_, err := GetProcess(pid)
-		if err == nil {
-			return errors.ErrProcessAlreadyRunning
-		}
+	// Check if the daemon is running
+	// TODO: Need some form of waiting period or timeout here before
+	// returning success. The process may die shortly after spawn.
+	_, err = GetProcess(pid)
```
Comment (kshlm) [inline daemon/daemon.go]: `pid` would be uninitialized here. I think you should be using  `cmd.Process.Pid`

File: daemon/daemon.go (hunk 1 above)
Comment (prashanthpai) [inline daemon/daemon.go]: Okay.

**Comment by prashanthpai** [comment]
@kshlm Have addressed review comments and updated the PR.

**Pull request merged**